kasa/discover.py

```diff
@@ -62,21 +68,49 @@ def do_discover(self) -> None:
         """Send number of discovery datagrams."""
         req = json.dumps(Discover.DISCOVERY_QUERY)
         _LOGGER.debug("[DISCOVERY] %s >> %s", self.target, Discover.DISCOVERY_QUERY)
-        encrypted_req = self.protocol.encrypt(req)
+        encrypted_req = TPLinkSmartHomeProtocol.encrypt(req)
+        new_req = binascii.unhexlify("020000010000000000000000463cb5d3")
         for i in range(self.discovery_packets):
             self.transport.sendto(encrypted_req[4:], self.target)  # type: ignore
+            self.transport.sendto(new_req, self.new_target)  # type: ignore
 
     def datagram_received(self, data, addr) -> None:
         """Handle discovery responses."""
         ip, port = addr
         if ip in self.discovered_devices:
             return
 
-        info = json.loads(self.protocol.decrypt(data))
+        if port == 9999:
+            info = json.loads(TPLinkSmartHomeProtocol.decrypt(data))
+            device_class = Discover._get_device_class(info)
+            device = device_class(ip)
+        else:
+            info = json.loads(data[16:])
+            device_class = Discover._get_new_device_class(info)
+            owner = Discover._get_new_owner(info)
+            if owner is not None:
+                owner_bin = bytes.fromhex(owner)
+
+            _LOGGER.debug(
+                "[DISCOVERY] Device owner is %s, empty owner is %s",
+                owner_bin,
+                self.emptyUser,
+            )
+            if owner is None or owner == "" or owner_bin == self.emptyUser:
+                _LOGGER.debug("[DISCOVERY] Device %s has no owner", ip)
+                device = device_class(ip, Auth())
+            elif (
+                self.authentication is not None
```

Comment:
The self.authentication needs to be already there if we get this far, so checking for the owner should be enough.